python/tvm/relay/frontend/sklearn.py

```diff
@@ -296,6 +298,174 @@ def _RobustPCA(op, inexpr, dshape, dtype, columns=None):
+    """
+
+    column_shape = [dshape[0], 1]
+
+    if not inverse:
+        lower_bound_x = _op.take(quantiles, indices=_expr.const(0))
+        upper_bound_x = _op.take(quantiles, indices=_expr.const(column_shape[0] - 1))
+        lower_bound_y = _expr.const(0)
+        upper_bound_y = _expr.const(1)
+    else:
+        lower_bound_y = _op.take(quantiles, indices=_expr.const(0))
+        upper_bound_y = _op.take(quantiles, indices=_expr.const(1))
+        lower_bound_x = _expr.const(0)
+        upper_bound_x = _expr.const(1)
+
+    x_shape = _infer_type(X_col).checked_type.shape
+    lower_bounds_idx = _op.equal(_op.cast(X_col, "float"), _op.broadcast_to(lower_bound_x, x_shape))
+    upper_bounds_idx = _op.equal(_op.cast(X_col, "float"), _op.broadcast_to(upper_bound_x, x_shape))
+
+    isfinite_mask = _op.logical_not(_op.isnan(X_col))
+    isfinite_mask_indices = all_indicies
+
+    X_col_finite = _op.take(X_col, _op.cast(isfinite_mask_indices, "int"))
+
+    if not inverse:
+        X_col_finite = _op.take(X_col_finite, indices=all_indicies)
+
+        interp1 = _op.interpolate(
+            X_col_finite,
+            _op.take(quantiles, indices=quantile_indicies),
+            _op.cast(references, "float32"),
+        )
+        interp2 = _op.interpolate(
+            _op.negative(X_col_finite),
+            _op.reverse(_op.negative(_op.take(quantiles, indices=quantile_indicies)), axis=0),
+            _op.reverse(_op.negative(_op.cast(references, "float32")), axis=0),
+        )
+
+        mul1 = _op.subtract(interp1, interp2)
+        mul_out = _op.multiply(_op.cast(_expr.const(0.5), "float"), _op.cast(mul1, "float"))
+    else:
+        interp_out = _op.interpolate(X_col_finite, qt.references_, quantiles)
+        mul_out = interp_out
+
+    mul_out = _op.reshape(mul_out, column_shape)
+    scatter_idx = _op.argwhere(
+        _op.squeeze(
+            _op.reshape(_op.cast(_op.take(isfinite_mask, all_indicies), "float32"), column_shape)
+        )
+    )
+    X_col = _op.scatter(_op.cast(X_col, "float32"), _op.cast(scatter_idx, "int"), mul_out, axis=0)
+
+    upper_bound_y = _op.full(upper_bound_y, (25, 1), dtype="float32")
+    upper_bounds_idx = _op.argwhere(_op.squeeze(_op.reshape(upper_bounds_idx, column_shape)))
+    X_col = _op.scatter(
+        _op.cast(X_col, "float32"), _op.cast(upper_bounds_idx, "int"), upper_bound_y, axis=0
+    )
+
+    lower_bound_y = _op.full(lower_bound_y, (25, 1), dtype="float32")
+    lower_bounds_idx = _op.argwhere(_op.squeeze(_op.reshape(lower_bounds_idx, column_shape)))
+    X_col = _op.scatter(
+        _op.cast(X_col, "float32"), _op.cast(lower_bounds_idx, "int"), lower_bound_y, axis=0
+    )
+
+    return X_col
+
+
+def _QuantileTransformer(op, inexpr, dshape, dtype, columns=None):
+    """
+    Scikit-Learn Transformer:
+    Transform features using quantiles information.
+    """
+    out = []
+    inverse = False
+
+    features = dshape[1]
+    all_indicies = list(range(dshape[0]))
```

Comment:
Yeah, I'd like a confirmation for this as well. And if there are any sample models that we must support. It is fine to have X_col have dynamic shape but the batch size is needed to iterate through to apply the column transformer. Might be hard to change that.

Reply:
I think Yunhan was able to use the tile op to do that dynamically. But I agree that it's a bit harder to do that and especially for this op already seems complicated enough.

Reply:
I'm not sure if tile would work in this case cause it only repeats data and doesn't allow applying a complex transformation. I made some changes though and think it will work for every case. I don't use dshape anymore, which I probably shouldn't have used as it represents the dshape to the model not a op in the pipeline. I allow the batch size to be dynamic and use shape_of to grab it. For columns, I use the number of quantiles which is an attribute of the op.